src/main/java/org/opentripplanner/routing/core/RoutingRequest.java

```diff
@@ -675,6 +688,11 @@ public void setBannedAgencies(String s) {
             bannedAgencies = new HashSet<String>(Arrays.asList(s.split(",")));
     }
 
+    public void setWhiteListedAgencies(String s) {
+        if (s != null && !s.equals(""))
+            whiteListedAgencies = new HashSet<String>(Arrays.asList(s.split(",")));
```

Comment:
This is a tiny micro-optimization but I'll mention it since it might add to readability: 
`Sets.newHashSet(s.split(",")`